src/node.cc

```diff
@@ -1159,6 +1153,32 @@ Maybe<bool> ProcessEmitDeprecationWarning(Environment* env,
                                    deprecation_code);
 }
 
+static void OnMessage(Local<Message> message, Local<Value> error) {
+  Isolate* isolate = Isolate::GetCurrent();
+  switch (message->ErrorLevel()) {
+    case Isolate::MessageErrorLevel::kMessageWarning: {
+      Environment* env = Environment::GetCurrent(isolate);
+      if (!env) {
+        break;
+      }
+      v8::String::Utf8Value filename(isolate,
+          message->GetScriptOrigin().ResourceName());
+      // (filename):(line) (message)
+      std::stringstream warning;
+      warning << *filename;
+      warning << ":";
+      warning << message->GetLineNumber(env->context()).FromMaybe(-1);
+      warning << " ";
+      v8::String::Utf8Value msg(isolate, message->Get());
+      warning << *msg;
```

Comment:
This isn't exactly wrong but we commonly use `snprintf()` on a stack-allocated buffer for this.

Reply:
i have no opinion one way or another but i would assume @refack does

Reply:
I like it that `snprintf()` takes a format string... (and yeah a stack buffer is nice too)
But it's up to you.

Reply:
how much should i allocate for the output buffer?

Reply:
I think the fact that the filename length is arbitrary makes that hard – 1024 bytes should be okay in practice, but you can also leave the code as it currently is

Reply:
i think i'll just leave it.